src/main.c

```diff
@@ -128,9 +128,17 @@ main (int argc, char *argv[])
   }
   while (next_option != -1);
 
-  bufchk (getenv ("HOME"), MP);
+  if (getenv ("HOME") == NULL)
+  {
+    fprintf (stderr, "Error: Environmental variable $HOME can't be used. Unable to determine home directory\n");
+    return 1;
+  }
+  
   char HOMEDIR[strlen (getenv ("HOME")) + 1];
```

Comment:
When the length of a string is known, I prefer to use it when declaring the size of an array. Although I'm using getenv() more than once, that's ok.